README.md

```diff
@@ -1304,113 +1406,1033 @@ public:
+             */
+            const serialization::pimpl::ObjectType getType() const;
+
+            /**
+             * Deserializes the underlying binary data and produces the object of type T.
+             *
+             * <b>CAUTION</b>: The type that you provide should be compatible with what object type is returned with
+             * the getType API, otherwise you will either get an exception of incorrectly try deserialize the binary data.
+             *
+             * @tparam T The type to be used for deserialization
+             * @return The object instance of type T.
+             */
+            template <typename T>
+            std::auto_ptr<T> get() const;
+```
+
+TypedData does late deserialization of the data only when the get method is called.
+
+This TypedData allows you to retrieve the data type of the underlying binary to be used when being deserialized. This class represents the type of a Hazelcast serializable object. The fields can take the following values:
+1. <b>Primitive types</b>: `factoryId=-1`, `classId=-1`, `typeId` is the type ID for that primitive as listed in
+`SerializationConstants`
+2. <b>Array of primitives</b>: `factoryId=-1`, `classId=-1`, `typeId` is the type ID for that array as listed in
+`SerializationConstants`
+3. <b>IdentifiedDataSerializable</b>: `factoryId`, `classId` and `typeId` are non-negative values as registered by
+the `DataSerializableFactory`.
+4. <b>Portable</b>: `factoryId`, `classId` and `typeId` are non-negative values as registered by the `PortableFactory`.
+5. <b>Custom serialized objects</b>: `factoryId=-1`, `classId=-1`, `typeId` is the non-negative type ID as 
+registered for the custom object. 
+
+
+# 8. Development and Testing
+
+Hazelcast C++ client is developed using C++. If you want to help with bug fixes, develop new features or
+tweak the implementation to your application's needs, you can follow the steps in this section.
+
+## 8.1. Building and Using Client From Sources
+
+For compiling with SSL support: 
+
+- You need to have the openssl (version 1.0.2) installed in your development environment: (i) Add openssl `include` directory to include directories, (ii) Add openssl `library` directory to the link directories list (This is the directory named `tls`. e.g. `cpp/Linux_64/hazelcast/lib/tls`), (iii) Set the openssl libraries to link.
+ 
+- You can provide your openssl installation directory to cmake using the following flags: 
+    - -DHZ_OPENSSL_INCLUDE_DIR="Path to open installation include directory" 
+    - -DHZ_OPENSSL_LIB_DIR="Path to openssl lib directory"
+
+- Use -DHZ_COMPILE_WITH_SSL=ON
+
+- Check the top level CMakeLists.txt file to see which libraries we link for openssl in which environment. 
+    - For Mac OS and Linux, we link with "ssl" and "crypto" 
+    - For Windows, install also the 1.1.x version of the openssl library. An example linkage for 64 bit library and release build: "libeay32MD ssleay32MD libcrypto64MD libSSL64MD".
```

Comment:
... of the OpenSSL library. An example linkage for 64-bit library and release build: "libeay32MD ssleay32MD libcrypto64MD libSSL64MD".